lib/logstash/inputs/log4j.rb

```diff
@@ -59,7 +57,25 @@ def register
     @logger.info("Log4j input")
   end # def register
 
-  public
+  def run(output_queue)
+    if server?
+      while !stop?
+        Thread.start(@server_socket.accept) do |s|
+          s.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
```

Comment:
This dynamic mixing in of behaviour to an instance will invalidate (J)Ruby's method lookup cache.  The runtime stops and rebuilds it.  A serious performance implication. Although it happens just once, we all need to know to use it only when there is no alternative.

Also the idiomatic way, is to do ```ruby s.extend(::LogStash::Util::SocketPeer)``` because instance.extend adds methods at  instance scope and class.extend adds them at class scope.

Consider using a delegating proxy.